pandas/core/dtypes/dtypes.py

```diff
@@ -871,7 +871,7 @@ def _parse_dtype_strict(cls, freq):
         raise ValueError("could not construct PeriodDtype")
 
     @classmethod
-    def construct_from_string(cls, string):
+    def construct_from_string(cls, string: str_type):
```

Comment:
the code looks like it accepts DateOffset

Reply:
Good point.  Maybe it shouldn't?

Reply:
reverted this change, punting on this for now